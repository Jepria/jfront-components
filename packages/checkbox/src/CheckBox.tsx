import React from "react"
import styled from "styled-components"
import nextId from "react-id-generator"

export interface CheckBoxInterface
  extends React.InputHTMLAttributes<HTMLInputElement> {
  /**
   * Подпись к чекбоксу
   */
  label?: string
  className?: string
  style?: React.CSSProperties
  orientation?: "left" | "right"
}

const StyledCheckBoxInput = styled.input.attrs({ type: "checkbox" })`
  cursor: pointer;
`

const StyledCheckBoxLabel = styled.label`
  box-sizing: border-box;
  padding: 2px;
  height: 20px;
  overflow: hidden;
  text-overflow: ellipsis;
  cursor: pointer;
  white-space: nowrap;
  width: 100%;
  text-align: left;
  -webkit-touch-callout: none; /* iOS Safari */
  -webkit-user-select: none; /* Safari */
  -khtml-user-select: none; /* Konqueror HTML */
  -moz-user-select: none; /* Old versions of Firefox */
  -ms-user-select: none; /* Internet Explorer/Edge */
  user-select: none; /* Non-prefixed version, currently
                              supported by Chrome, Edge, Opera and Firefox */
`
interface CheckboxContainer {
  orientation?: "left" | "right"
}

const StyledCheckBox = styled.div<CheckboxContainer>`
  box-sizing: border-box;
  padding: 2px;
  display: flex;
  align-items: center;
  ${(props) =>
    props.orientation && props.orientation === "left"
      ? `flex-direction: row-reverse;`
      : `flex-direction: row;`};
  width: 100%;
  cursor: pointer;
  font-size: ${(props) => props.theme.fontSize.md};
  font-family: ${(props) => props.theme.fontFamily};
  color: ${(props) => props.theme.checkbox.color};
  &:hover {
    background: ${(props) => props.theme.checkbox.hoverBgColor};
    color: ${(props) => props.theme.checkbox.hoverColor};
  }
`

StyledCheckBox.defaultProps = {
  theme: {
    fontSize: {
      md: "12px",
    },
    fontFamily: "tahoma, arial, helvetica, sans-serif",
    checkbox: {
      hoverBgColor: "#eee",
      hoverColor: "#000",
      color: "#000",
    },
  },
}

export const CheckBox = React.forwardRef<HTMLInputElement, CheckBoxInterface>(
  ({ id, label, className, style, type, orientation, ...props }, ref) => {
    const htmlId = id ? id : nextId()

    return (
      <StyledCheckBox
        className={className}
        style={style}
        orientation={orientation}
      >
        {label && (
          <StyledCheckBoxLabel
            htmlFor={htmlId}
            onDoubleClick={(e) => {
              /** IE fix checkbox double-click issue **/
              if ((document as any).documentMode) {
                e.stopPropagation()
                e.currentTarget.click()
              }
            }}
          >
            {label}
          </StyledCheckBoxLabel>
        )}
        <StyledCheckBoxInput
          id={htmlId}
          ref={ref}
          onDoubleClick={(e) => {
            /** IE fix checkbox double-click issue **/
            if ((document as any).documentMode) {
              e.stopPropagation()
              e.currentTarget.click()
            }
          }}
          {...props}
        />
      </StyledCheckBox>
    )
  },
)
