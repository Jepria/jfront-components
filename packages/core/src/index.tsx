export * from '@jfront-ui/app'
export * from '@jfront-ui/datepicker'
export * from '@jfront-ui/form'
//export * from '@jfront-ui/grid'
export * from '@jfront-ui/hooks'
//export * from '@jfront-ui/input'
export * from '@jfront-ui/layout'
//export * from '@jfront-ui/pagingbar'
//export * from '@jfront-ui/tabpanel'
//export * from '@jfront-ui/toolbar'
