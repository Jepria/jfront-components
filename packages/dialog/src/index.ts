export * from './SimpleDialog'
export * from './AlertDialog'
export * from './FormDialog'
export * from './ErrorDialog'
export * from './WarningDialog'