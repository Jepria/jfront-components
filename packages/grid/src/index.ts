export * from './common'
export * from './jepgrid'
