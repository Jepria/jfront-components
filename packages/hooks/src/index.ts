export * from "./useOnClickOutside"
export * from "./useRouteMapping"
export * from "./useTree/"
export * from "./useTree/types"
export * from "./useFilter"
