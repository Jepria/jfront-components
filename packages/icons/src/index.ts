export * from "./Add"
export * from "./Delete"
export * from "./Edit"
export * from "./Exclamation"
export * from "./First"
export * from "./Last"
export * from "./Loading"
export * from "./Next"
export * from "./OpenIcon"
export * from "./Prev"
export * from "./Refresh"
export * from "./Save"
export * from "./Search"
export * from "./SettingsSmall"
export * from "./Split"
export * from "./View"
