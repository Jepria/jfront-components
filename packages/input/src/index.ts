export interface InputProps {
  error?: string
  name?: string
  isLoading?: boolean
}

export * from "./SelectInput"
export * from "./TextInput"
export * from "./NumberInput"
export * from "./MaskedInput"
export * from "./MoneyInput"
export * from "./DecimalInput"
export * from "./MoneyInput"
