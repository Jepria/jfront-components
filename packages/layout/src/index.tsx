import styled from 'styled-components';

export const FormContainer = styled.div`
  padding: 5px;
`;

export const VerticalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const HorizontalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
`;

export const Page = styled.div`
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const Header = styled.header`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;

export const Content = styled.section`
  box-sizing: border-box;
  width: 100%;
  overflow: auto;
  -webkit-box-flex: 1;
      -ms-flex: 1;
          flex: 1;
  -ms-flex-positive: 1;
      flex-grow: 1;
`;

export const Footer = styled.footer`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;
