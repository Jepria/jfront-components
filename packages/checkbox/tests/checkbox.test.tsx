import { fireEvent, render, screen } from "@testing-library/react"
import React from "react"
import { CheckBox } from "../src"

test("Checking for the existence of an element CheckBox", () => {
  expect(render(<CheckBox />))
})

test("Checking CheckBox clicking", () => {
  let checkBoxState = false
  render(
    <CheckBox
      label="label2"
      checked={checkBoxState}
      onChange={(checked) => {
        checkBoxState = checked.target.checked
      }}
    />,
  )
  fireEvent.click(screen.getByLabelText("label2"))
  expect(checkBoxState).toEqual(true)
})

test("Matches snapshot ", () => {
  const { asFragment } = render(<CheckBox />)
  expect(asFragment()).toMatchSnapshot()
})

test("Checkbox renders correctly", () => {
  const tools = render(<CheckBox label="CheckBox Name" />)
  expect(tools.asFragment()).toMatchSnapshot()
})
