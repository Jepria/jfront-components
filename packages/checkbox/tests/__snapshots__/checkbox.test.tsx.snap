// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
<DocumentFragment>
  <span
    class="sc-AxiKw bsspJY"
  >
    <label
      class="sc-AxirZ ieqohX"
      for="id4"
    >
      CheckBox Name
    </label>
    <input
      class="sc-AxjAm jIaAfJ"
      id="id4"
      type="checkbox"
      value=""
    />
  </span>
</DocumentFragment>
`;

exports[`Matches snapshot  1`] = `
<DocumentFragment>
  <span
    class="sc-AxiKw bsspJY"
  >
    <label
      class="sc-AxirZ ieqohX"
      for="id3"
    />
    <input
      class="sc-AxjAm jIaAfJ"
      id="id3"
      type="checkbox"
      value=""
    />
  </span>
</DocumentFragment>
`;
